File: magic-link-ng-dotnet/ClientApp/src/app/models/connection.ts
export class ConnectionInfo {
    constructor(
    ) {}

    public accountName: string;
    public password: string;
    public subscriptionKey: string;
    public jwtToken: string = null;
    public rootURL: string = "public.mago.cloud";
    public ui_culture: string;
    public culture: string;
    public isDebugEnv: boolean;
}

export class LoginRequest {
    public url: string;
    public accountName: string;
    public password: string;
    public subscriptionKey: string;
}

export class TbUserData {
    constructor(
    ) {}

    public token: string;
    public userName: string;
    public password: string;
    public subscriptionKey: string;
    public isLogged: boolean;
}

export class SetDataResponse {
    public xmlData: string;
    public warnings: string;
}

File: openecommerce-ng-dotnet/Models/Connection.cs
using Microarea.Tbf.Model.API;

namespace openecommerce_ng_dotnet.Models
{
    public class LoginRequest {
        public string? url {get; set;}
        public string? accountName {get; set;}
        public string? password {get; set;}
        public string? subscriptionKey {get; set;}
        public bool overwrite {get; set;}
        public string? appId {get; set;}
    }

    public class SetDataResponse {
        public string? xmlData {get; set;}
        public string? warnings {get; set;}
    }

    public interface IMagoConnection {
        public MagoAPIClient? APIClient {get; set; }
        public TbUserData? TbUserData {get; set; }
    }

    public class MagoConnection : IMagoConnection {
        public MagoAPIClient? _APIclient = null;
        public TbUserData? _TbUserData = null;
        public MagoAPIClient? APIClient {get { return _APIclient; }  set { _APIclient = value; } }
        public TbUserData? TbUserData {get { return _TbUserData; } set{ _TbUserData = value; } }
    }

    public class TbUserDataWrapper : TbUserData
    {
        public TbUserDataWrapper() : base("","","","") {}
    }

}
